[r] Compiler nag on 'pessimizing move' from `nanoarrow_testing.hpp`

After syncing with the headers as in `dist/` now (and from a very gentle and gingerly use of a little JSON as now supported) I started to see this in a little test repo:

```
nanoarrow/nanoarrow_testing.hpp: In member function ‘ArrowErrorCode nanoarrow::testing::TestingJSONWriter::WriteColumn(std::ostream&, const ArrowSchema*, const ArrowArrayView*)’:
nanoarrow/nanoarrow_testing.hpp:470:42: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
  470 |                                 std::move(dictionary_output.str()));
      |                                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~                               
nanoarrow/nanoarrow_testing.hpp:470:42: note: remove ‘std::move’ call
```

and indeed, just removing the `std::move()`  appears to work in the 'well `valgrind` does not complain' sense.

So I can send you a PR that does the equivalent of this if you want me to.

I am on Ubuntu 23.10 with `g++-13` version 13.2.0 in case it matters.

--- a/src/nanoarrow/nanoarrow_testing.hpp
+++ b/src/nanoarrow/nanoarrow_testing.hpp
@@ -467,7 +467,7 @@ class TestingJSONWriter {
       NANOARROW_RETURN_NOT_OK(
           WriteColumn(dictionary_output, field->dictionary, value->dictionary));
       dictionaries_.RecordArray(field->dictionary, value->dictionary->length,
-                                std::move(dictionary_output.str()));
+                                dictionary_output.str());
     }
 
     return NANOARROW_OK;

